### DISCUSSION TITLE
Modifying web UI

### DISCUSSION THREAD

**Discussion by jetpax** [dzurikmiroslav/esp32-evse#23] [Q&A]
If I understand correctly, the webUI is compressed in web.cpio, and the contents are minified

Is there a source for this so that the UI can be modified?

**Comment by dzurikmiroslav** [comment]
Yes exactly.
I don't like to mix different projects (cmake & npm) in one git repository, I would like to solve it as a git submodule or something similar... 
The repository is not available yet, I have no idea how to refer to the CI result from another repository.

**Comment by jetpax** [comment]
Hi Miroslav,

 Totally get that.  I've been working on using rjwats/esp8266-react for my own EV project, but you seem to have really thought through the issues like dynamic config vs. user settings, scripting, good api structure etc., so I look forward to seeing your approach when youre ready

I have a few questions, tho,

1. Why did you choose Bootstrap over MUI?
2. Did Preact gain you much over standard React?
3. Do you plan to use Websockets  e.g. for streaming data from the 'back-end'?
4. What build system did you use CRA or Vite  or other?
5. Why did you choose Berry  Scripting over say Microvium or Duktape?

And thanks for sharing  a great project btw!

**Reply by dzurikmiroslav** [reply]
1. Previously WebUI was vanilla JS with bootstrap library. So I keep design and implement it with react-boostrap. I'm not against MUI, but it would me take more time...
2. Final bundle is more smaller than react, also using preact-router which has smaller footprint
3. I had an attempt :-) lwIP stack has limited max socket count (16 - for all system), in default setup socket support omitted. Especially with websockets there is a problem with various proxies and port-forwarding
4. I use preact-cli
5. Duktape has bigger memory footprint than Berry. I did not detailed study Microvium, but C library can only execute bytecode not source code? Berry also uses Tasmota, that convinced me the most. But personally, I more like JS syntax than Berry - Lua like :-)

**Comment by iboldurev** [comment]


**Reply by dzurikmiroslav** [reply]
Hi, which additional functionality ?

**Reply by dzurikmiroslav** [reply]
I can't make the repository public because it contains know-how from the company where I am employed.

**Reply by dzurikmiroslav** [reply]
Hi, refactoring the entire web interface is not a simple task... Sorry, but I currently don't have the time and resources to refactor and publish these source codes in the near future. I'm rather focusing on new functionality such as integration with ThingBoard and the OCPP protocol...
